src/platform/datapath_raw_socket.c

```diff
@@ -408,21 +441,23 @@ CxPlatResolveRoute(
         Operation->Context = Context;
         Operation->Callback = Callback;
         Operation->PathId = PathId;
+        Route->State = RouteResolving;
         CxPlatDispatchLockAcquire(&Worker->Lock);
         CxPlatListInsertTail(&Worker->Operations, &Operation->WorkerLink);
```

Comment:
We should have a code comment here explaining why we suddenly go through this hypercomplicated code path when we discover we need to send an NS. Just something like "Neighbor solicitation requires an exchange of network packets. To keep from blocking the entire connection work queue, which is shared with other connections, queue an operation on the dedicated route resolution worker to do the solicitation."

Reply:
done